### PR TITLE
Update "Azure AD" references to "Microsoft Entra ID"

### PR DESCRIPTION
**Pull request by wojciechcloudkubed**
# Resolves [#3650](https://github.com/microsoft/AzureTRE/issues/3650)

## What is being addressed

Update "Azure AD" references to "Microsoft Entra ID" 

## Acceptance criteria

- [ ] Documentation all updated
- [ ] Microsoft Entra Workforce tenant used rather than Azure AD Tenant
- [ ] Code files containing comments updated

### CODE REVIEW

**Comment by marrobi** [comment]
@wojciechcloudkubed still one here? `ui/README.md`

File: ui/README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ The UI was built using Create React App and Microsoft Fluent UI. Further details
 ## Run the UI
 - Ensure `deploy_ui=false` is not set in your `./config.yaml` file
 - In the root of the repo, run `make tre-deploy`. This will provision the necessary resources in Azure, build and deploy the UI to Azure blob storage, behind the App Gateway used for the API. The deployment process will also create the necessary `config.json`, using the `config.source.json` as a template.
-- In Azure AD, locate the TRE Client Apps app (possibly called Swagger App). In the Authentication section add reply URIs for:
+- In Microsoft Entra Workforce ID, locate the TRE Client Apps app (possibly called Swagger App). In the Authentication section add reply URIs for:
```
Comment (marrobi) [inline ui/README.md:10]: Not sure need Workforce? Workforce is used when referring to a Workforce Tenant vs Customer Tenant

```suggestion
- In Microsoft Entra ID, locate the TRE Client Apps app (possibly called Swagger App). In the Authentication section add reply URIs for:
```

File: Makefile [hunk 2]
```diff
@@ -362,8 +362,8 @@ setup-local-debugging: ## 🛠️ Setup local debugging
 	&& . ${MAKEFILE_DIR}/devops/scripts/load_env.sh ${MAKEFILE_DIR}/core/private.env \
 	&& . ${MAKEFILE_DIR}/devops/scripts/setup_local_debugging.sh
 
-auth: ## 🔐 Create the necessary Azure Active Directory assets
-	$(call target_title,"Setting up Azure Active Directory") \
+auth: ## 🔐 Create the necessary Microsoft Entra Workforce ID assets
```
Comment (marrobi) [inline Makefile:365]: I think need to do a find and replace for "Microsoft Entra Workforce ID" -> Microsoft Entra ID.

File: cli/tre/commands/login.py [hunk 3]
```diff
@@ -43,7 +43,7 @@ def login():
               + 'https://<id>.<location>.cloudapp.azure.com/')
 @click.option('--client-id',
               required=False,
-              help='The Client ID of the Azure AD application for the API (optional for API versions >= v0.5.7)')
+              help='The Client ID of the Microsoft Entra Workforce ID application for the API (optional for API versions >= v0.5.7)')
 @click.option('--aad-tenant-id',
               required=False,
               help='The Tenant ID for the AAD tenant to authenticate with (optional for API versions >= v0.5.7)')
```
Comment (marrobi) [inline cli/tre/commands/login.py:49]:
```suggestion
              help='The Tenant ID for the Microsoft Entra ID tenant to authenticate with (optional for API versions >= v0.5.7)')
```

**Review by marrobi — COMMENTED** [review]
Think need to refrain from using Workforce, other then when discuss tenants. Are lots of references to AAD that need changing to Entra ID.

**Review by marrobi — CHANGES_REQUESTED** [review]

**Review by marrobi — APPROVED** [review]
LGTM - can you create another issue - much lower priority to update references to Azure AD in the code base, including variable names is scripts, python code, UI code etc. 

Thanks.

**Review by tim-allen-ck — APPROVED** [review]
LGTM

**Pull request merged**